pages/accountLists/[accountListId].page.tsx

```diff
@@ -80,59 +77,48 @@ const AccountListIdPage = ({
   );
 };
 
-export const getServerSideProps: GetServerSideProps = async (context) => {
-  const session = await getSession(context);
-  const apiToken = session?.user.apiToken;
-  // If no token from session, redirect to login page
-  if (!apiToken) {
-    return {
-      redirect: {
-        destination: '/login',
-        permanent: false,
-      },
-    };
-  }
+export const getServerSideProps = makeGetServerSideProps(
+  async (session, context) => {
+    try {
+      const { query } = context;
+      if (typeof query.accountListId !== 'string') {
+        throw new Error('Invalid accountListId');
+      }
 
-  try {
-    const { query } = context;
-    if (typeof query.accountListId !== 'string') {
-      throw new Error('Invalid accountListId');
-    }
-
-    const ssrClient = makeSsrClient(apiToken);
-    const { data } = await ssrClient.query<
-      GetDashboardQuery,
-      GetDashboardQueryVariables
-    >({
-      query: GetDashboardDocument,
-      variables: {
-        accountListId: query.accountListId,
-        // TODO: implement these variables in query
-        // endOfDay: DateTime.local().endOf('day').toISO(),
-        // today: DateTime.local().endOf('day').toISODate(),
-        // twoWeeksFromNow: DateTime.local()
-        //   .endOf('day')
-        //   .plus({ weeks: 2 })
-        //   .toISODate(),
-      },
-    });
+      const ssrClient = makeSsrClient(session.user.apiToken);
+      const { data } = await ssrClient.query<
+        GetDashboardQuery,
+        GetDashboardQueryVariables
+      >({
+        query: GetDashboardDocument,
+        variables: {
+          accountListId: query.accountListId,
+          // TODO: implement these variables in query
+          // endOfDay: DateTime.local().endOf('day').toISO(),
+          // today: DateTime.local().endOf('day').toISODate(),
+          // twoWeeksFromNow: DateTime.local()
+          //   .endOf('day')
+          //   .plus({ weeks: 2 })
+          //   .toISODate(),
+        },
+      });
 
-    return {
-      props: {
-        data,
-        modal: query?.modal?.toString() ?? '',
-        session,
```

Comment:
was there a reason we were passing the session through here? I feel like there was but I can't remember, maybe it was due to fetching the API Token?

Reply:
It's used by `_app.page.tsx` to initialize the session provider. All pages that require a login need to return the `session` prop.